network/p2pNetwork.go

```diff
@@ -216,10 +237,19 @@ func (n *P2PNetwork) Relay(ctx context.Context, tag protocol.Tag, data []byte, w
 func (n *P2PNetwork) Disconnect(badnode Peer) {
 	switch node := badnode.(type) {
```

Comment:
nit: this entire function could be rewritten from switch to type assertion and logging `n.log.Warnf("Unknown peer type %T", badnode)` if not `peer.ID`